src/main/scala/firrtl/backends/proto/ProtoBufEmitter.scala

```diff
@@ -59,10 +62,47 @@ sealed abstract class ProtoBufEmitter(prereqs: Seq[TransformDependency])
   override def optionalPrerequisiteOf = Seq.empty
   override def invalidates(a: Transform) = false
 
-  override def execute(state: CircuitState) =
-    state.copy(annotations = state.annotations :+ Annotation.ProtoBufSerialization(state.circuit, Some(outputSuffix)))
+  private def emitAllModules(circuit: Circuit): Seq[Annotation.ProtoBufSerialization] = {
+    // For a given module, returns a Seq of all modules instantited inside of it
+    def collectInstantiatedModules(mod: Module, map: Map[String, DefModule]): Seq[DefModule] = {
+      // Use list instead of set to maintain order
+      val modules = ArrayBuffer.empty[DefModule]
+      def onStmt(stmt: Statement): Unit = stmt match {
+        case DefInstance(_, _, name, _) => modules += map(name)
+        case _: WDefInstanceConnector => throwInternalError(s"unrecognized statement: $stmt")
+        case other => other.foreach(onStmt)
+      }
+      onStmt(mod.body)
+      modules.distinct.toSeq
+    }
+    val modMap = circuit.modules.map(m => m.name -> m).toMap
+    // Turn each module into it's own circuit with it as the top and all instantied modules as ExtModules
+    circuit.modules.collect {
+      case m: Module =>
+        val instModules = collectInstantiatedModules(m, modMap)
+        val extModules = instModules.map {
+          case Module(info, name, ports, _) => ExtModule(info, name, ports, name, Seq.empty)
+          case ext: ExtModule => ext
+        }
+        val newCircuit = Circuit(m.info, extModules :+ m, m.name)
+        Annotation.ProtoBufSerialization(newCircuit, Some(outputSuffix))
+    }
+  }
```

Comment:
Given that this is almost a copy-paste of the same method in `FirrtlEmitter`, can we instead factor them out into a common utility that both call? https://github.com/chipsalliance/firrtl/blob/de56a19b0a240e39366dc2d979ec05c65e0ada63/src/main/scala/firrtl/backends/firrtl/FirrtlEmitter.scala#L22

Preferable `private[firrtl]` so that we don't leak it into the public API but still would be better to not have the code duplication.